types/sinon/index.d.ts

```diff
@@ -1628,7 +1628,7 @@ declare namespace Sinon {
          * replacement can be any value, including spies, stubs and fakes.
          * This method only works on non-accessor properties, for replacing accessors, use sandbox.replaceGetter() and sandbox.replaceSetter().
          */
-        replace<T, TKey extends keyof T>(obj: T, prop: TKey, replacement: T[TKey]): T[TKey];
+        replace<T, TKey extends keyof T, R extends T[TKey]>(obj: T, prop: TKey, replacement: R): R;
```

Comment:
I think you should add a default type argument for the new type parameter, for it to be backwards-compatible with calls that explicitly passed type arguments.
So `replace<T, TKey extends keyof T, R extends T[TKey]>` should be `replace<T, TKey extends keyof T, R extends T[TKey] = T[TKey]>`.

Reply:
Thank you, I've missed this.